compiler/rustc_hir_analysis/src/collect/predicates_of.rs

```diff
@@ -62,6 +62,143 @@ pub(super) fn predicates_of(tcx: TyCtxt<'_>, def_id: DefId) -> ty::GenericPredic
+                    bug!("unexpected {node:?}")
+                };
+
+                let trait_generics = tcx.generics_of(assoc_item.trait_item_def_id.unwrap());
+                for (arg, dup_def) in bidir_outlives {
+                    let orig_region = icx.astconv().ast_region_to_region(arg, None);
+                    if !matches!(orig_region.kind(), ty::ReEarlyBound(..)) {
+                        // Only early-bound regions can point to the original generic parameter.
+                        continue;
+                    }
+
+                    let Node::Lifetime(hir::Lifetime { res: hir::LifetimeName::Param(orig_def), .. }) = tcx.hir().get(arg.hir_id) else { bug!() };
+                    let Some(orig_trait_index) = trait_generics.param_def_id_to_index(tcx, orig_def.to_def_id()) else { bug!() };
+                    let Some(dup_trait_index) = trait_generics.param_def_id_to_index(tcx, dup_def.to_def_id()) else { bug!() };
+
+                    let impl_generics = tcx.generics_of(def_id);
+                    let orig_impl_index = orig_trait_index - 1;
+                    let dup_impl_index = dup_trait_index - trait_generics.parent_count as u32
+                        + impl_generics.parent_count as u32;
+                    let orig_impl_def =
+                        impl_generics.param_at(orig_impl_index as usize, tcx).def_id;
+                    let dup_impl_def = impl_generics.param_at(dup_impl_index as usize, tcx).def_id;
+
+                    let orig_region = ty::Region::new_early_bound(
+                        tcx,
+                        ty::EarlyBoundRegion {
+                            def_id: orig_impl_def,
+                            index: orig_impl_index,
+                            name: arg.ident.name,
+                        },
+                    );
+                    let dup_region = ty::Region::new_early_bound(
+                        tcx,
+                        ty::EarlyBoundRegion {
+                            def_id: dup_impl_def,
+                            index: dup_impl_index,
+                            name: arg.ident.name,
+                        },
+                    );
+
+                    predicates.push((
+                        ty::Binder::dummy(ty::PredicateKind::Clause(ty::Clause::RegionOutlives(
+                            ty::OutlivesPredicate(orig_region, dup_region),
+                        )))
+                        .to_predicate(icx.tcx),
+                        arg.ident.span,
+                    ));
+                    predicates.push((
+                        ty::Binder::dummy(ty::PredicateKind::Clause(ty::Clause::RegionOutlives(
+                            ty::OutlivesPredicate(dup_region, orig_region),
+                        )))
+                        .to_predicate(icx.tcx),
+                        arg.ident.span,
+                    ));
+                }
+                debug!(?predicates);
```

Comment:
predicates in this side could be predicates_of(trait_gat) with identity substs of the method rebased onto the trait ref of the impl + renumbered lifetimes specific to the opaques